apps/wakunode2/app.nim

```diff
@@ -44,8 +43,40 @@ import
 when defined(rln):
   import ../../waku/v2/protocol/waku_rln_relay
 
+logScope:
+  topics = "wakunode app"
 
-type AppResult[T] = Result[T, string]
+
+# Git version in git describe format (defined at compile time)
+const git_version* {.strdefine.} = "n/a"
+
+type
+  App* = object
+    version: string
+    conf: WakuNodeConf
+
+    rng: ref HmacDrbgContext
+    peerStore: Option[WakuPeerStorage]
+    archiveDriver: Option[ArchiveDriver]
+    archiveRetentionPolicy: Option[RetentionPolicy]
```

Comment:
Makes sense to me. But I prefer addressing it in a future PR instead of as part of this refactoring. Otherwise, rebasing the feature branch can get messy.